source/diagnostic.cpp

```diff
@@ -37,7 +37,7 @@ spv_diagnostic spvDiagnosticCreate(const spv_position position,
   diagnostic->position = *position;
   diagnostic->isTextSource = false;
   memset(diagnostic->error, 0, length);
-  strncpy(diagnostic->error, message, length);
+  memcpy(diagnostic->error, message, length);
```

Comment:
What's wrong with the strncpy?  

Now that I read the documentation for strncpy, the memset on the previous line is redundant with the strncpy.

Reply:
I've made an update to use strncpy again and removed the memset line which fixes the error I was getting trying to compile.

Reply:
The first version was right : I also get the stringop-truncation truncation error unless I use memcpy instead of strncpy.
You're right the memset is useless.
Also the last char of the string must be set to 0.
```
diagnostic->error,[length] = 0;
memcpy(diagnostic->error, message, length);
```

Reply:
Since the destination is allocated one larger than the source and zeroed out via `memcpy`, I think that replacing the `strncpy` with `strcpy` will suffice.

Reply:
it does not prevent from overflow though...